sklearn/impute/_base.py

```diff
@@ -465,11 +465,72 @@ def transform(self, X):
             n_missing = np.sum(mask, axis=0)
             values = np.repeat(valid_statistics, n_missing)
             coordinates = np.where(mask.transpose())[::-1]
-
             X[coordinates] = values
 
         return super()._concatenate_indicator(X, X_indicator)
 
+    def inverse_transform(self, X):
+        """Convert the data back to the original representation.
+
+        Inverts the `transform` operation performed on an array.
+        This operation can only be performed after :class:`SimpleImputer` is
+        instantiated with `add_indicator=True`.
+
+        Note that ``inverse_transform`` can only invert the transform in
+        features that have binary indicators for missing values. If a feature
+        has no missing values at ``fit`` time, the feature won't have a binary
+        indicator, and the imputation done at ``transform`` time won't be
+        inverted.
+
+        Parameters
+        ----------
+        X : array-like of shape \
+                (n_samples, n_features + n_features_missing_indicator)
+            The imputed data to be reverted to original data. It has to be
+            an augmented array of imputed data and the missing indicator mask.
+
+        Returns
+        -------
+        original_X : ndarray, shape (n_samples, n_features)
+            The original X with missing values as it was prior
+            to imputation.
+        """
+        check_is_fitted(self)
+
+        if not self.add_indicator:
+            raise ValueError("'inverse_transform' works only when "
+                             "'SimpleImputer' is instantiated with "
+                             "'add_indicator=True'. Got 'add_indicator={}' "
+                             "instead.".format(self.add_indicator))
+
+        missing_feature_count = len(self.indicator_.features_)
+
+        # Split the augmented array into imputed array and its missing
+        # indicator mask.
+        feature_count = X.shape[1] - missing_feature_count
+        imputed_arr = X[:, :feature_count].copy()
+        missing_mask = X[:, feature_count:].astype(np.bool)
+        orig_cols = len(self.statistics_)
```

Comment:
```suggestion
        n_features_original = len(self.statistics_)
```

Reply:
What is the difference with what you called `feature_count` originally?

Reply:
`feature_count` is the count of features that not completely empty. And `orig_cols` is the original number of features (including empty features). Changed their names to make it clearer.